### PR TITLE
KAN-10 Cards Table Redesign

### PR DESCRIPTION
**Pull request by XimenaPulidoGutierrez**
###  Description
The design of the cards table redesign has been updated

### Solution
I created a second version of almost everything so as not to interrupt the flow of the other codes. (Tag, Tags, CutomTagCard, PageTittleCards) and I modified the Strings to corrected the word cards and the routes to point to new routes.

I modified the "PageTittleCards" so that it changes according to the primary color, and I corrected the "customTagCards" so that only the text is visible instead of the background color.

### Notes
Mr. Fausto mentioned a correction related to the use of the word "Card." I made the change in the section where it was used and which I was responsible for redesigning. However, I noticed that in elements like "Card Interface," which is a component that is used but not displayed, the word "Card" is still being used. I’m not sure if I should also modify that. Since it is widely used in the system, I decided not to make additional changes to avoid creating problems where none exist.

Also, my ticket mentioned that I should check if the search bar works. However, as I mentioned in the WhatsApp group, the search bar only works with letters.

### Tickets
[TICKET] (https://one-sm.atlassian.net/browse/KAN-10)

### Screenshots / Screencasts
image

### Screenshots UPDATED
Captura de pantalla 2024-12-19 170622
Captura de pantalla 2024-12-19 170634
Captura de pantalla 2024-12-19 170644
Captura de pantalla 2024-12-19 170652
Captura de pantalla 2024-12-19 171509
Captura de pantalla 2024-12-19 171521

### CODE REVIEW

File: src/components/PageTitleCards.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+interface Props {
+  mainText: string;
+  subText?: string;
+  primaryColor: string;
+}
+
+const PageTitleCards = ({ mainText, subText, primaryColor }: Props) => {
```
Comment (imadiaz) [inline src/components/PageTitleCards.tsx:7]: can we rename this to "PageTitleCard"?

Comment (XimenaPulidoGutierrez) [reply src/components/PageTitleCards.tsx:7]: "PageTittleTag" is okay? Following the rules of "Card = Tag"?

Comment (XimenaPulidoGutierrez) [reply src/components/PageTitleCards.tsx]: Change made

**Comment by XimenaPulidoGutierrez** [comment]
> can u also update the screen shots to be sure that this design is responsive in a different screen sizes?

I updated the screenshots.

File: src/components/PageTitleTag.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import {theme } from 'antd';
+
+const { useToken } = theme;
+
+interface Props {
+  mainText: string;
+  subText?: string;
+}
+const PageTitleCards: React.FC<Props> = ({ mainText, subText }) => {
+  const { token } = useToken();  // Se obtiene el token en PageTitleCards
```
Comment (imadiaz) [inline src/components/PageTitleTag.tsx:11]: can you translate the comments to english?

File: src/components/PageTitleTag.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  // Usamos el color primario del tema
```
Comment (imadiaz) [inline src/components/PageTitleTag.tsx:13]: same here

File: src/pages/card/components/Tag.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  localAdminCardDetails,
+  sysAdminCardDetails,
+} from "../../routes/Routes";
+
+interface CardProps {
+  data: CardInterface;
+  rol: UserRoles;
+}
+
+const InformationPanel = ({ data, rol }: CardProps) => {
```
Comment (imadiaz) [inline src/pages/card/components/Tag.tsx:29]: should we rename this? i mean this is Tag, why we use informationpanel?

File: src/components/PageTitleTag.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import {theme } from 'antd';
+
+const { useToken } = theme;
+
+interface Props {
+  mainText: string;
+  subText?: string;
+}
+const PageTitleCards: React.FC<Props> = ({ mainText, subText }) => {
```
Comment (imadiaz) [inline src/components/PageTitleTag.tsx:10]: should we use PageTitleCard instead of PageTitleCards?

Comment (XimenaPulidoGutierrez) [reply src/components/PageTitleTag.tsx]: I change the name to "PageTittleTag" to be as the same as the name of the component

File: src/pages/card/components/Tag.tsx (hunk 4 above)
Comment (XimenaPulidoGutierrez) [reply src/pages/card/components/Tag.tsx]: Yes you are right. I made the change now the const is "Tag" Is that okay?

File: src/components/PageTitleTag.tsx (hunk 3 above)
Comment (XimenaPulidoGutierrez) [reply src/components/PageTitleTag.tsx]: Changes made

File: src/components/PageTitleTag.tsx (hunk 2 above)
Comment (XimenaPulidoGutierrez) [reply src/components/PageTitleTag.tsx]: Yes

File: src/pages/card/components/Tag.tsx (hunk 4 above)
Comment (imadiaz) [reply src/pages/card/components/Tag.tsx]: Yes

File: src/components/PageTitleTag.tsx (hunk 5 above)
Comment (imadiaz) [reply src/components/PageTitleTag.tsx]: Make sense

File: src/pages/card/Tags.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+import InformationPanel from "./components/Tag";
+import { CardInterface } from "../../data/card/card";
+import { UserRoles } from "../../utils/Extensions";
+import { UnauthorizedRoute } from "../../utils/Routes";
+
+interface CardsProps {
+  rol: UserRoles;
+}
+
+const Cards = ({ rol }: CardsProps) => {
```
Comment (imadiaz) [inline src/pages/card/Tags.tsx:18]: should we rename also this to Tags right?

**Review by imadiaz — COMMENTED** [review]
just one more comment

File: src/pages/card/Tags.tsx (hunk 6 above)
Comment (XimenaPulidoGutierrez) [reply src/pages/card/Tags.tsx]: Yes. correction made

File: src/pages/card/components/Tag.tsx (hunk 4 above)
Comment (XimenaPulidoGutierrez) [reply src/pages/card/components/Tag.tsx]: Change alredy madre

**Review by imadiaz — APPROVED** [review]

**Review by ErickGS28 — APPROVED** [review]
lgtm

**Pull request merged**